assets/svelte/components/SelectedElementFloatingMenu/DragMenuOption.svelte

```diff
@@ -137,24 +137,38 @@
     return $dragElementInfo.parentElementClone.children.item($dragElementInfo.selectedIndex)
   }
 
-  function findHoveredSiblingIndex(dragDirection: DragDirection, e: MouseEvent) {
-    // TODO: This detection is not very intuitive. We should detect some % of element overlap (30% maybe) instead.
+  // Considers that the dragged element is hovering another one if their overlap is more than 50%.
+  function findHoveredSiblingIndex(dragDirection: DragDirection, mouseDiff: Coords, e: MouseEvent) {
+    const draggedElementInfo = $dragElementInfo.siblingLocationInfos[$dragElementInfo.selectedIndex]
     if (dragDirection === "vertical") {
-      return $dragElementInfo.siblingLocationInfos.findIndex(
-        (rect) => rect.top < e.y && rect.bottom + rect.marginBottom > e.y,
-      )
+      const { top, y, bottom, ...rest } = draggedElementInfo
+      const draggedRect = { ...rest, y: y + mouseDiff.y, top: top + mouseDiff.y, bottom: bottom + mouseDiff.y }
+      return $dragElementInfo.siblingLocationInfos.findIndex((rect, index) => {
+        if (index !== $dragElementInfo.selectedIndex) {
+          const overlap = Math.max(0, Math.min(draggedRect.bottom, rect.bottom) - Math.max(draggedRect.top, rect.top))
+          const horizontalOverlapRatio = overlap / Math.min(draggedRect.height, rect.height)
```

Comment:
Hold on, I just noticed I named this `horizontalOverlapRatio` when it's actually vertical

Reply:
Fixed